src/main/java/ch/njol/skript/events/EvtWorldInit.java

```diff
@@ -0,0 +1,65 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import org.bukkit.World;
+import org.bukkit.event.Event;
+import org.bukkit.event.world.WorldInitEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Arrays;
+
+public class EvtWorldInit extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("World Init", EvtWorldInit.class, WorldInitEvent.class, "world init[ialization] [of %-worlds%]")
+			.description("Called when a world is initialised. As all default worlds are initialised before",
+				"any scripts are loaded, this event is only called for newly created worlds.",
+				"World management plugins might change the behaviour of this event though.")
+			.examples("on world init of \"world_the_end\":")
+			.since("1.0, INSERT VERSION (defining worlds)");
+	}
+
+	@Nullable
```

Comment:
Actually I think this doesn't need to be nullable. If you set the `[of %-worlds%]` to `[of %worlds%]` it should grab the event-value world of the event, and then it'll just be checking the event world with the event world, meaning a gurenteed pass, similar to return true. so the null check in the check(event) methods can be removed and the nullable annotations on these events.

Reply:
From the short little test I've done it doesn't seem do want to do that, instead when the check is called `worlds` is just null and causes a runtime error, probably is best to be left alone, or I'm doing something wrong

Reply:
From the testing I've been able to do with this I was unable to get it to work, so it's possible I've done something wrong knowing me but for now I think it's fine I could always revisit it in the future.